server/lib/nl/utils.py

```diff
@@ -578,6 +580,8 @@ def place_detection_with_heuristics(query_fn, query: str) -> List[str]:
     logging.info(f"Trying place detection with: {q}")
     try:
       for p in query_fn(q):
+        # remove any stop words like "a", "the" etc from the place.
```

Comment:
sorry, is this needed? what bug is this fixing?

Reply:
added a comment to explain. it was detecting "the SF bay area" and we have a special case for "SF bay area". so it would think these are two places and first use "the SF bay area" (the longer one) for Maps API query which would find "san francisco" and not the one we want (for which we have a DCID). I have also seen this happen with "the United States" and while that does not cause trouble, it makes sense to remove "the" "a" "an" etc from the place names detected.

Reply:
discussed offline, just removing "the" instead of all stop words. Can revisit this later.